apis/v1alpha1/httproute_types.go

```diff
@@ -59,6 +59,15 @@ type HTTPRouteHost struct {
 	// +optional
 	Hostnames []string `json:"hostnames,omitempty"`
 
+	// TLS defines the TLS certificate to use for Hostnames defined in this
+	// Route. The certificate defined here takes effect only if the AllowedInRoute
+	// field is set to true in the associated Gateway resource.
+	//
+	// Support: Core
+	//
+	// +optional
+	TLS *TLSInRouteConfig `json:"tls" protobuf:"bytes,2,opt,name=tls"`
```

Comment:
Sorry I don't have access to the doc to get more context - shouldn't this equally apply to tcp route as well?

Reply:
It doesn't really make sense at the TCPRoute level because TCPRouteMatch doesn't have any SNI based matching semantics.
This might make sense in TLSRoute though.